### DISCUSSION TITLE
✔️ Towers will need an update

### DISCUSSION THREAD

**Discussion by kevingrillet** [zebscripts/AFK-Daily#72] [Upcoming AFK changes]
With the version [1.65](https://www.afkarena.net/update-1-65-patch-notes#New_Celestial_Tower): 
```
NEW CELESTIAL TOWER
Added a new Celestial Tower called the “Celestial Sanctum“.
Players can unlock it after completing Stage 29-60 of the Campaign. 
```

But we need to wait before we know what will require an update...

**Comment by Zebiano** [comment]
It will most likely require an update I can imagine. They will probably reorganize the tower tiles. If they implement a scroll thing that would be nice, as at least the script wouldn't break.

Also, since it's a pretty high level actually (basically campaign 30), this will also be very niche. Though I can imagine some users of the script are there already, which kinda means we'd need to implement logic for the script to decide if it should fight there or not.

There's various option to this approach:
1. We simply code the script in a way that it just pretends the tower is there and fights it. We have to ensure that everything goes well in case there isn't one though.
2. We test for a HEX value to check if the tower exists, and depending on a variable, we fight it or not.
3. We have a variable to check if we should fight or not, and leave the responsibility to the user, as we won't check for a HEX. By default, this variable will be set to `false`.
4. We implement some sort of system that knows VIP and Campaign levels and acts upon those things.

Now, realistically speaking you might have to join some of those approaches to get the best one (for example join 4 and 2). Also, I'd of course love to have the 4th implemented, as this would open up the possibilities for lower leveled users. But I'm just not sure how to implement it yet, I've thought about this for a long time already. Ideas are welcome of course.

So... What conclusion do we come to? Ideally, a whole system would be built, but that seems out of this world and it'll take a long time to implement, as well as maintain. Which means we will most likely run with one of the first three options. Which one we pick will depend on the way this tower gets implemented. 

We'll have to wait to find out.

~~Also, I'm not a big fan of creating an issue that isn't even an issue yet. I wonder where the best place for this would be... Probably in the discussions part of the repository. I just didn't find any category that fits. I'll check later if I can create custom categories.~~ Discussion created!

 Thank you for the heads-up with this potential issue!

**Reply by kevingrillet** [reply]
So, the patch is here:

- Slow transition before entering the King's Tower page 3-5s
- Slow transition before entering a King tower 1-2s (more a freeze than a transition)
- New disposition of the different towers

![Bluestacks_2021-06-04_15-02_SRgGM](https://user-images.githubusercontent.com/7203617/121172223-c2bfed00-c857-11eb-8be4-038553b1588f.png)
![Bluestacks_2021-06-08_12-45_08L7P](https://user-images.githubusercontent.com/7203617/121172231-c489b080-c857-11eb-9807-818cf445b4e7.png)

**Reply by kevingrillet** [reply]
It's added at my PR

**Reply by kevingrillet** [reply]
Important notes:
- You need to have a team
- You need to beat Floor 1 manually
